crates/nu-cmd-lang/src/core_commands/do_.rs

```diff
@@ -69,6 +69,33 @@ impl Command for Do {
         let block: Closure = call.req(engine_state, caller_stack, 0)?;
         let rest: Vec<Value> = call.rest(engine_state, caller_stack, 1)?;
         let ignore_all_errors = call.has_flag(engine_state, caller_stack, "ignore-errors")?;
+
+        if call.has_flag(engine_state, caller_stack, "ignore-shell-errors")? {
+            nu_protocol::report_shell_error(
```

Comment:
Slightly related to the conversation in #14331, but should this (and the `report_shell_error` below) be `report_shell_warning` instead? It doesn't seem to be used anywhere else, but this does seem like the kind of thing which is more of a "warning" than an "error". It looks like the only difference is printing "Warning" instead of "Error"?

Reply:
Good catch!  I change it :-)